tap_gitlab/streams.py

```diff
@@ -2,11 +2,24 @@
 
 from typing import Any, Dict, Optional
 
-from tap_gitlab.client import GitLabStream, ProjectBasedStream, GroupBasedStream
+from singer_sdk import typing as th  # JSON Schema typing helpers
+
+from tap_gitlab.client import GitLabStream, GroupBasedStream, ProjectBasedStream
 from tap_gitlab.transforms import object_array_to_id_array, pop_nested_id
 
 # Project-Specific Streams
 
+# all user objects in non-user streams follow the same structure
```

Comment:
We could imagine having a `base_objects.py` to store this and others. And do something similr in tap-github to imho. @aaronsteers do you have any guidance / good examples on how this can be structured? Any existing implementation come to mind?

Reply:
I've not noticed other "sub-objects" that are reused as much as the users are, but if we find them, I think it would make sense to do the same. I should have thought of doing this for `tap-github` before :)

Reply:
For reference - https://github.com/MeltanoLabs/tap-github/issues/68